### ISSUE TITLE
Help! Possible environmental problems running Spark Batch

### ISSUE THREAD

**Issue by alexbfree** [tomncooper/CSC8101-Documentation#25]
Hi.. I hope someone is still online. I am having some issues with code that was previously running that is now not running, even though I changed nothing.... 

After training the model and getting recommendations, (which return OK), I try and process them:
```
recommendedList = sorted(recoms, key=lambda x: x[2], reverse=True)[:10]

for i in range(0,len(recommendedList)):
    print("%2d. %s" % (i+1,getNameIfKnown(str(recoms[i][1]))))
```

This gives the following error:
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-85-4a2301522f78> in <module>()
      1 for i in range(0,len(recommendedList)):
----> 2     print("%2d. %s" % (i+1,getNameIfKnown(str(recoms[i][1]))))

<ipython-input-82-68d4ea0df5ac> in getNameIfKnown(netflixID)
      1 # get the canonical movie name, if we matched it - otherwise Netflix title
      2 def getNameIfKnown(netflixID):
----> 3     mapping = broadcastMovieIDMapping.value
      4     if netflixID in mapping:
      5         return mapping[netflixID]

/home/ubuntu/spark/spark-2.1.0-bin-hadoop2.7/python/pyspark/broadcast.py in value(self)
    104         """
    105         if not hasattr(self, "_value") and self._path is not None:
--> 106             self._value = self.load(self._path)
    107         return self._value
    108 

/home/ubuntu/spark/spark-2.1.0-bin-hadoop2.7/python/pyspark/broadcast.py in load(self, path)
     90 
     91     def load(self, path):
---> 92         with open(path, 'rb', 1 << 20) as f:
     93             # pickle.load() may create lots of objects, disable GC
     94             # temporary for better performance

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/spark-ee05fb8f-6d3d-499e-b790-68a68865ff28/pyspark-d49a7466-bbdd-47bc-95ef-6df5e3e999d9/tmpyipq7sd8'
```

Not sure if it's related, but during ALS training I am also getting in the console
```
[Stage 115:==================================================>    (34 + 2) / 37]17/03/02 21:16:26 WARN MemoryStore: Not enough space to cache rdd_148_35 in memory! (computed 16.0 MB so far)
17/03/02 21:16:26 WARN MemoryStore: Not enough space to cache rdd_148_36 in memory! (computed 15.7 MB so far)
[Stage 115:====================================================>  (35 + 2) / 37]17/03/02 21:16:27 WARN MemoryStore: Not enough space to cache rdd_148_36 in memory! (computed 15.7 MB so far)
17/03/02 21:16:49 WARN Executor: 1 block locks were not released by TID = 959:
[rdd_257_0]
17/03/02 21:16:50 WARN Executor: 1 block locks were not released by TID = 960:
[rdd_258_0]
17/03/02 21:16:50 WARN Executor: 1 block locks were not released by TID = 961:
[rdd_257_0]
17/03/02 21:16:50 WARN Executor: 1 block locks were not released by TID = 962:
[rdd_258_0]
```

How can I get up and running again?

**Comment by alexbfree** [comment]
OK this is non-urgent now, I found a workaround (just save to file rather than trying to print it out first). Still curious if anyone does have a clue what is going on, but it's not blocking me any more.

**Comment by PetoMichalak** [comment]
Hi,

regarding the out of memory problem, have you already tweaked driver memory and executor memory parameters as mentioned in [Task 0](https://github.com/tomncooper/CSC8101-Documentation/blob/master/spark/spark-coursework-spec.md#task-0)? This seems to be a similar problem as successfully resolved in [issue #2 ](https://github.com/tomncooper/CSC8101-Documentation/issues/2).
